Personality.java

```diff
@@ -0,0 +1,161 @@
+ */
+
+public class Personality {
+    /* Constants:
+     * int DIMENSIONS - Total number of personality dimensions in Keiser Temperment
+     * char A and B - Used for finding answers of each type, A or B.  There are also - answers, but these are not used.
+     * Stringp[] PERSONAOPTIONS - Array of personality trait identifying letters.
+     * XPERSONASTR - When personality is neither of the options and receives to X
+     */
+    public final static int DIMENSIONS = 4;
+    public final static char A = 'A';
+    public final static char B = 'B';
+    public static final String[] PERSONAOPTIONS = {"E", "I", "S", "N", "T", "F", "J", "P"};
+    public static final String XPERSONASTR = "X";
+
+    public static void main(String[] args) throws FileNotFoundException {
+        introduction();
+        System.out.println("Input file name?");
+        Scanner console = new Scanner(System.in);
+        Scanner personalityData = new Scanner(new File(console.next()));
+        System.out.print("Output file name?");
+        PrintStream processedData = new PrintStream(console.next());
+        while (personalityData.hasNextLine()) {
+            processedData.print(personalityData.nextLine() + ": ");
+            String[] answersByTrait = getAnswersByTrait(personalityData.nextLine());
+            int[] percentBTraits = getPercentBPerTrait(getCountsOfChar(answersByTrait, A), getCountsOfChar(answersByTrait, B));
+            processedData.print(Arrays.toString(percentBTraits));
+            processedData.print(" = ");
+            processedData.print(getPersonalityType(percentBTraits));
+            processedData.println();
+        }
+    }
+
+    /* @fn introduction
+     * Introduces program to user.
+     */
+    public static void introduction() {
+        System.out.println("This program processes a file of answers to the");
+        System.out.println("Keirsey Temperament Sorter. It converts the");
+        System.out.println("various A and B answers for each person into");
+        System.out.println("a sequence of B-percentages and then into a");
+        System.out.println("four-letter personality type.");
+    }
+
+    /* @fn getAnswersByTrait
+     * Sorts answers into one of the DIMENSIONS strings for each traits based on answer location in the array
+     *
+     * @param answersString contains all answers for a single person
+     * @return String[DIMENSIONS] including a string for all the answers of the DIMENSIONS traits
+     */
+    public static String[] getAnswersByTrait(String answersString) {
+        String[] answersByTrait = new String[DIMENSIONS];
+        int arrayLoc = 0;
+        for (int iter = 0; iter < answersString.length(); iter++) {
+            if (arrayLoc > 6) {
+                arrayLoc = 0;
+            }
+            //To index to 0, 1, 1, 2, 2, 3, 3 due to order of answers for traits
+            int arrayIndex = Math.round((arrayLoc + 1) / 2);
+            arrayLoc++;
```

Comment:
simplify by like do the > 6 here and use %.  ie arrayLoc = (araryLoc+1)%7  (for [0,6]